From 98f10ef05c1f817827af6559d4976c5e77e62d30 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 19:24:45 +0530
Subject: [PATCH 097/281] OpenWrt: 214-spidev_h_portability.patch

kernel: fix linux/spi/spidev.h portability issues with musl

Felix will try to get this define included into musl

lede-commit: 795e7cf60de19e7a076a46874fab7bb88b43bbff
Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: If8e5ad0908757a70eb2202d1009b48de84479fe2
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 include/uapi/linux/spi/spidev.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/spi/spidev.h b/include/uapi/linux/spi/spidev.h
index 0c3da08f2aff..2a6d51008256 100644
--- a/include/uapi/linux/spi/spidev.h
+++ b/include/uapi/linux/spi/spidev.h
@@ -93,7 +93,7 @@ struct spi_ioc_transfer {
 
 /* not all platforms use <asm-generic/ioctl.h> or _IOC_TYPECHECK() ... */
 #define SPI_MSGSIZE(N) \
-	((((N)*(sizeof (struct spi_ioc_transfer))) < (1 << _IOC_SIZEBITS)) \
+	((((N)*(sizeof (struct spi_ioc_transfer))) < (1 << 13)) \
 		? ((N)*(sizeof (struct spi_ioc_transfer))) : 0)
 #define SPI_IOC_MESSAGE(N) _IOW(SPI_IOC_MAGIC, 0, char[SPI_MSGSIZE(N)])
 
-- 
2.17.1

